From 4f5e34004b51ec16fd3bbafd566f012989f03fc5 Mon Sep 17 00:00:00 2001
From: YunQiang Su <yunqiang@isrc.iscas.ac.cn>
Date: Fri, 15 May 2026 15:13:18 +0800
Subject: [PATCH] Add -openruyi-linux to X86_64Triples and RISCV64Triples

Some applications like rustc may call clang with option
  --target=riscv64-unknown-linux-gnu
and complains it cannot find libraries.
Origin: https://github.com/Sakura286/openRuyi/blob/a8c497cd4c6a05f4158075d1ad3b1edfe8a7b76c/SPECS/llvm22/2001-Add-openruyi-linux-to-X86_64Triples-and-RISCV64Tripl.patch
Applied to the bundled LLVM after moving it into llvm/.

Rebased with diff against the pinned LLVM 24 source.

--- a/clang/lib/Driver/ToolChains/Gnu.cpp
+++ b/clang/lib/Driver/ToolChains/Gnu.cpp
@@ -2365,7 +2365,8 @@
       "x86_64-pc-linux-gnu",    "x86_64-redhat-linux6E",
       "x86_64-redhat-linux",    "x86_64-suse-linux",
       "x86_64-manbo-linux-gnu", "x86_64-slackware-linux",
-      "x86_64-unknown-linux",   "x86_64-amazon-linux"};
+      "x86_64-unknown-linux",   "x86_64-amazon-linux",
+      "x86_64-openruyi-linux"};
   static const char *const X32Triples[] = {"x86_64-linux-gnux32",
                                            "x86_64-pc-linux-gnux32"};
   static const char *const X32LibDirs[] = {"/libx32", "/lib"};
@@ -2441,7 +2442,8 @@
                                                "riscv32-unknown-elf"};
   static const char *const RISCV64LibDirs[] = {"/lib64", "/lib"};
   static const char *const RISCV64Triples[] = {"riscv64-unknown-linux-gnu",
-                                               "riscv64-unknown-elf"};
+                                               "riscv64-unknown-elf",
+                                               "riscv64-openruyi-linux"};
 
   static const char *const RISCV32beLibDirs[] = {"/lib32", "/lib"};
   static const char *const RISCV32beTriples[] = {"riscv32be-unknown-linux-gnu",
